modules/history/src/views/full/MessageTaskBar.tsx

```diff
@@ -0,0 +1,116 @@
+import { Divider, Icon } from '@blueprintjs/core'
+import { AnchorButton, Colors, Position, Tooltip } from '@blueprintjs/core'
+import { Toolbar } from 'botpress/ui'
+import React, { Fragment } from 'react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+
+import { MessageGroup } from '../../typings'
+
+import style from './style.scss'
+import { MessageDownload } from './MessageDownload'
+
+interface Filters {
+  flag: boolean
+}
+
+interface Props {
+  currentConv: string
+  updateFilters: (f: Filters) => void
+  useAsFilter: boolean
+  selectedCount: number
+  flag: () => void
+  unflag: () => void
+  messageGroups: MessageGroup[]
+}
+
+interface State {
+  filters: Filters
+  currentConv: string
+  showCopiedTooltip: boolean
+}
+
+export class MessageTaskBar extends React.Component<Props, State> {
+  state = {
+    filters: {
+      flag: false
+    },
+    currentConv: undefined,
+    showCopiedTooltip: false
+  }
+
+  componentDidUpdate() {
+    if (this.props.currentConv !== this.state.currentConv) {
+      this.setState({ currentConv: this.props.currentConv, filters: { flag: false } })
+    }
+  }
+
+  toggleFlagFilter = () => {
+    this.state.filters.flag = !this.state.filters.flag
```

Comment:
Never mutate state directly, always use `this.setState({ filters })`

If the reason you did it this way is because you had a race condition, you could have done:

```js
this.setState({ filters: { ...this.state.filters, flag: !this.state.filters } }, () => {
  this.props.updateFilters(this.state.filters)
})
```

But I don't understand why it's a state variable, you don't seem to reference it anywhere except here.  If it doesn't impact the rendering, make it a variable.  If it *does* impact rendering, never mutate it directly.

Reply:
You're right, filters doesn't belong in the state. It is old code that I copied from another file.